src/hooks/useBackButton.tsx

```diff
@@ -17,7 +22,10 @@ export default function useBackButton({ username }: Props) {
         return;
       }
 
-      if (canGoBack) {
+      // If the user has a different page in their history such as the Edit Collection Page, don't use back()
+      const dontGoBack = historyStack[historyStack.length - 2]?.includes('/edit?collectionId=');
```

Comment:
nit: would prefer this variable defined outside of the callback, similar to `canGoBack`. the reason is because this forrces `historyStack` to be a dependency of this callback, which is an array that's almost guaranteed to change more frequently than a simple true/false.

e.g.
```js
const previousRoute = historyStack[historyStack.length - 2]
const backDisabled = previousRoute?.includes('/edit?collectionId=');

const handleBackClick = useCallback(() => {
  ...
  if (!canGoBack || backDisabled) { <----- changed the logic a bit, not sure if better or worse
    void replace(`/${username}`);
    return;
  }
  back();
}, [...backDisabled...])
```

Reply:
oh yeah totally!

Reply:
done ✅